pkg/tsdb/cloudwatch/log_query.go

```diff
@@ -202,8 +202,20 @@ func groupResults(results *data.Frame, groupingFieldNames []string, removeNonTim
 			newFrame := results.EmptyCopy()
 			newFrame.Name = groupKey
 			newFrame.Meta = results.Meta
-			// remove grouping indices
-			newFrame.Fields = removeFieldsByIndex(newFrame.Fields, removeFieldIndices)
+			if fromSyncQuery {
```

Comment:
how come it has to only be on sync queries?

Reply:
I could do it on both, but it gets a bit weird if I do it on non-sync queries. When I fixed the parsing to remove the naming fields, I only did it in the case of sync queries (which was maybe a mistake) so if I set the display name for non-sync queries it ends up sort of doubling those names.
CloudWatch logs query without expression
without DisplayNameFromDS:
![Screenshot 2023-09-08 at 10 50 19 AM](https://github.com/grafana/grafana/assets/5421859/d04b0af3-e0da-4b4a-9cb7-064b9781e39e)

with DisplayNameFromDS:
![Screenshot 2023-09-08 at 10 49 01 AM](https://github.com/grafana/grafana/assets/5421859/c41b3bc2-9895-4b39-b7db-0aadb9ef698f)